Parsing error: Invalid value for lib provided: es2021

<!--
Please don't ignore this template.

If you ignore it, we're just going to respond asking you to fill it out, which wastes everyone's time.
The more relevant information you can include, the faster we can find the issue and fix it without asking you for more info.
-->

<!--
🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

This issue template is only for problems specifically with the `@typescript-eslint/parser` package.
That means you should only use this template if there are problems parsing some code.

If you have a problem with a specific lint rule, please back out and select the `@typescript-eslint/eslint-plugin` template.
-->

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

<!--
Include a ***minimal*** reproduction case.
The more irrelevant code/config you give, the harder it is for us to investigate.

Please consider creating an isolated reproduction repo to make it easy for the volunteer maintainers debug your issue.
-->

```js
module.exports = {
  root: true,
  parser: "@typescript-eslint/parser",
  parserOptions: {
    ecmaVersion: 2021,
    project: null,
  },
  plugins: ["@typescript-eslint"],
  rules: {
    "@typescript-eslint/ban-types": "error",
  },
};
```

```ts
// test.ts
export {}
```

<!--
Also include your tsconfig, if you're using type-aware linting
-->

```json
{
  "compilerOptions": {
    "module": "es2020",
    "target": "es2020",
    "lib": ["es2020", "dom"],
    "strict": true,
    "esModuleInterop": true,
    "moduleResolution": "node",
    "types": [],
    "forceConsistentCasingInFileNames": true,
  },
  "files": ["./test.ts"],
  "include": [],
  "exclude": ["node_modules/"]
}
```

**Expected Result**

<!--
What did you expect to happen?
Please be specific here - list the exact lines and messages you expect.
-->
ESLint executes successfully with no error.

**Actual Result**

<!--
What actually happened?
Please be specific here - list the exact lines and messages that caused errors
-->

```
/home/user/project/test.ts
  0:0  error  Parsing error: Invalid value for lib provided: es2021

✖ 1 problem (1 error, 0 warnings)
```

**Additional Info**

<!--
Did eslint throw an exception?

Please run your lint again with the --debug flag, and dump the output below.
i.e. eslint --ext ".ts,.js" src --debug
-->

<details><summary>eslint --debug dump</summary>
<p>

```
2020-09-01T19:40:17.809Z eslint:cli CLI args: [ '--config', '.eslintrc.test.js', '--ext', '.ts', './test.ts', '--debug' ]
2020-09-01T19:40:17.811Z eslint:cli Running on files
2020-09-01T19:40:17.821Z eslint:config-array-factory Loading .eslintignore file: /home/gabriel/Repos/leapfrog/.eslintignore
2020-09-01T19:40:17.822Z eslint:config-array-factory Loading JS config file: /home/gabriel/Repos/leapfrog/.eslintrc.test.js
2020-09-01T19:40:17.823Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '/home/gabriel/Repos/leapfrog', loose: false } ]
2020-09-01T19:40:17.825Z eslint:ignore-pattern   processed: { basePath: '/home/gabriel/Repos/leapfrog', patterns: [ '/**/node_modules/*' ] }
2020-09-01T19:40:17.825Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '/home/gabriel/Repos/leapfrog', loose: false } ]
2020-09-01T19:40:17.825Z eslint:ignore-pattern   processed: { basePath: '/home/gabriel/Repos/leapfrog', patterns: [ '/**/node_modules/*' ] }
2020-09-01T19:40:17.826Z eslint:file-enumerator Start to iterate files: [ './test.ts' ]
2020-09-01T19:40:17.826Z eslint:file-enumerator File: /home/gabriel/Repos/leapfrog/test.ts
2020-09-01T19:40:17.826Z eslint:cascading-config-array-factory Load config files for /home/gabriel/Repos/leapfrog.
2020-09-01T19:40:17.827Z eslint:cascading-config-array-factory No cache found: /home/gabriel/Repos/leapfrog.
2020-09-01T19:40:17.827Z eslint:config-array-factory Loading JS config file: /home/gabriel/Repos/leapfrog/.eslintrc.js
2020-09-01T19:40:17.827Z eslint:config-array-factory Config file found: /home/gabriel/Repos/leapfrog/.eslintrc.js
2020-09-01T19:40:17.827Z eslint:config-array-factory Loading {extends:"@gcangussu/eslint-config"} relative to /home/gabriel/Repos/leapfrog/.eslintrc.js
2020-09-01T19:40:17.829Z eslint:config-array-factory package.json was not found: Package subpath './package.json' is not defined by "exports" in /home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-configpackage.json
2020-09-01T19:40:17.829Z eslint:config-array-factory Loaded: @gcangussu/eslint-config (/home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/index.js)
2020-09-01T19:40:17.829Z eslint:config-array-factory Loading JS config file: /home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/index.js
2020-09-01T19:40:17.833Z eslint:config-array-factory Loading plugin "unicorn" from /home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/index.js
2020-09-01T19:40:17.833Z eslint:config-array-factory Loaded: eslint-plugin-unicorn@21.0.0 (/home/gabriel/Repos/leapfrog/node_modules/eslint-plugin-unicorn/index.js)
2020-09-01T19:40:17.927Z eslint:config-array-factory Plugin /home/gabriel/Repos/leapfrog/node_modules/eslint-plugin-unicorn/index.js loaded in: 94ms
2020-09-01T19:40:17.927Z eslint:config-array-factory Loading plugin "import" from /home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/index.js
2020-09-01T19:40:17.928Z eslint:config-array-factory Loaded: eslint-plugin-import@2.22.0 (/home/gabriel/Repos/leapfrog/node_modules/eslint-plugin-import/lib/index.js)
2020-09-01T19:40:17.982Z eslint:config-array-factory Plugin /home/gabriel/Repos/leapfrog/node_modules/eslint-plugin-import/lib/index.js loaded in: 54ms
2020-09-01T19:40:17.983Z eslint:config-array-factory Loading parser "@typescript-eslint/parser" from /home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/index.js
2020-09-01T19:40:17.984Z eslint:config-array-factory Loaded: @typescript-eslint/parser@4.0.1 (/home/gabriel/Repos/leapfrog/node_modules/@typescript-eslint/parser/dist/index.js)
2020-09-01T19:40:18.214Z eslint:config-array-factory Loading plugin "@typescript-eslint" from /home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/index.js
2020-09-01T19:40:18.215Z eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@4.0.1 (/home/gabriel/Repos/leapfrog/node_modules/@typescript-eslint/eslint-plugin/dist/index.js)
2020-09-01T19:40:18.335Z eslint:config-array-factory Plugin /home/gabriel/Repos/leapfrog/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 120ms
2020-09-01T19:40:18.335Z eslint:config-array-factory Loading {extends:"@gcangussu/eslint-config/quick"} relative to /home/gabriel/Repos/leapfrog/.eslintrc.js
2020-09-01T19:40:18.336Z eslint:config-array-factory package.json was not found: Package subpath './quick/package.json' is not defined by "exports" in /home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-configpackage.json
2020-09-01T19:40:18.336Z eslint:config-array-factory Loaded: @gcangussu/eslint-config/quick (/home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/quick.js)
2020-09-01T19:40:18.336Z eslint:config-array-factory Loading JS config file: /home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/quick.js
2020-09-01T19:40:18.337Z eslint:cascading-config-array-factory Stop traversing because of 'root:true'.
2020-09-01T19:40:18.338Z eslint:rules Loading rule 'array-callback-return' (remaining=281)
2020-09-01T19:40:18.339Z eslint:rules Loading rule 'block-scoped-var' (remaining=280)
2020-09-01T19:40:18.340Z eslint:rules Loading rule 'class-methods-use-this' (remaining=279)
2020-09-01T19:40:18.341Z eslint:rules Loading rule 'consistent-return' (remaining=278)
2020-09-01T19:40:18.342Z eslint:rules Loading rule 'constructor-super' (remaining=277)
2020-09-01T19:40:18.343Z eslint:rules Loading rule 'default-case' (remaining=276)
2020-09-01T19:40:18.343Z eslint:rules Loading rule 'default-param-last' (remaining=275)
2020-09-01T19:40:18.344Z eslint:rules Loading rule 'eqeqeq' (remaining=274)
2020-09-01T19:40:18.346Z eslint:rules Loading rule 'for-direction' (remaining=273)
2020-09-01T19:40:18.346Z eslint:rules Loading rule 'func-names' (remaining=272)
2020-09-01T19:40:18.347Z eslint:rules Loading rule 'getter-return' (remaining=271)
2020-09-01T19:40:18.348Z eslint:rules Loading rule 'grouped-accessor-pairs' (remaining=270)
2020-09-01T19:40:18.349Z eslint:rules Loading rule 'guard-for-in' (remaining=269)
2020-09-01T19:40:18.349Z eslint:rules Loading rule 'new-cap' (remaining=268)
2020-09-01T19:40:18.351Z eslint:rules Loading rule 'new-parens' (remaining=267)
2020-09-01T19:40:18.352Z eslint:rules Loading rule 'no-alert' (remaining=266)
2020-09-01T19:40:18.352Z eslint:rules Loading rule 'no-array-constructor' (remaining=265)
2020-09-01T19:40:18.352Z eslint:rules Loading rule 'no-async-promise-executor' (remaining=264)
2020-09-01T19:40:18.353Z eslint:rules Loading rule 'no-buffer-constructor' (remaining=263)
2020-09-01T19:40:18.353Z eslint:rules Loading rule 'no-caller' (remaining=262)
2020-09-01T19:40:18.353Z eslint:rules Loading rule 'no-case-declarations' (remaining=261)
2020-09-01T19:40:18.354Z eslint:rules Loading rule 'no-class-assign' (remaining=260)
2020-09-01T19:40:18.354Z eslint:rules Loading rule 'no-compare-neg-zero' (remaining=259)
2020-09-01T19:40:18.354Z eslint:rules Loading rule 'no-cond-assign' (remaining=258)
2020-09-01T19:40:18.355Z eslint:rules Loading rule 'no-console' (remaining=257)
2020-09-01T19:40:18.356Z eslint:rules Loading rule 'no-const-assign' (remaining=256)
2020-09-01T19:40:18.356Z eslint:rules Loading rule 'no-constant-condition' (remaining=255)
2020-09-01T19:40:18.357Z eslint:rules Loading rule 'no-constructor-return' (remaining=254)
2020-09-01T19:40:18.358Z eslint:rules Loading rule 'no-control-regex' (remaining=253)
2020-09-01T19:40:18.359Z eslint:rules Loading rule 'no-debugger' (remaining=252)
2020-09-01T19:40:18.359Z eslint:rules Loading rule 'no-delete-var' (remaining=251)
2020-09-01T19:40:18.359Z eslint:rules Loading rule 'no-dupe-args' (remaining=250)
2020-09-01T19:40:18.360Z eslint:rules Loading rule 'no-dupe-class-members' (remaining=249)
2020-09-01T19:40:18.360Z eslint:rules Loading rule 'no-dupe-else-if' (remaining=248)
2020-09-01T19:40:18.360Z eslint:rules Loading rule 'no-dupe-keys' (remaining=247)
2020-09-01T19:40:18.360Z eslint:rules Loading rule 'no-duplicate-case' (remaining=246)
2020-09-01T19:40:18.361Z eslint:rules Loading rule 'no-duplicate-imports' (remaining=245)
2020-09-01T19:40:18.362Z eslint:rules Loading rule 'no-else-return' (remaining=244)
2020-09-01T19:40:18.363Z eslint:rules Loading rule 'no-empty' (remaining=243)
2020-09-01T19:40:18.364Z eslint:rules Loading rule 'no-empty-character-class' (remaining=242)
2020-09-01T19:40:18.364Z eslint:rules Loading rule 'no-empty-function' (remaining=241)
2020-09-01T19:40:18.365Z eslint:rules Loading rule 'no-empty-pattern' (remaining=240)
2020-09-01T19:40:18.365Z eslint:rules Loading rule 'no-eval' (remaining=239)
2020-09-01T19:40:18.366Z eslint:rules Loading rule 'no-ex-assign' (remaining=238)
2020-09-01T19:40:18.366Z eslint:rules Loading rule 'no-extend-native' (remaining=237)
2020-09-01T19:40:18.368Z eslint:rules Loading rule 'no-extra-bind' (remaining=236)
2020-09-01T19:40:18.369Z eslint:rules Loading rule 'no-extra-boolean-cast' (remaining=235)
2020-09-01T19:40:18.370Z eslint:rules Loading rule 'no-extra-label' (remaining=234)
2020-09-01T19:40:18.370Z eslint:rules Loading rule 'no-fallthrough' (remaining=233)
2020-09-01T19:40:18.371Z eslint:rules Loading rule 'no-floating-decimal' (remaining=232)
2020-09-01T19:40:18.371Z eslint:rules Loading rule 'no-func-assign' (remaining=231)
2020-09-01T19:40:18.371Z eslint:rules Loading rule 'no-global-assign' (remaining=230)
2020-09-01T19:40:18.372Z eslint:rules Loading rule 'no-implicit-coercion' (remaining=229)
2020-09-01T19:40:18.373Z eslint:rules Loading rule 'no-implied-eval' (remaining=228)
2020-09-01T19:40:18.373Z eslint:rules Loading rule 'no-import-assign' (remaining=227)
2020-09-01T19:40:18.374Z eslint:rules Loading rule 'no-inner-declarations' (remaining=226)
2020-09-01T19:40:18.375Z eslint:rules Loading rule 'no-invalid-regexp' (remaining=225)
2020-09-01T19:40:18.376Z eslint:rules Loading rule 'no-irregular-whitespace' (remaining=224)
2020-09-01T19:40:18.377Z eslint:rules Loading rule 'no-iterator' (remaining=223)
2020-09-01T19:40:18.377Z eslint:rules Loading rule 'no-label-var' (remaining=222)
2020-09-01T19:40:18.378Z eslint:rules Loading rule 'no-labels' (remaining=221)
2020-09-01T19:40:18.378Z eslint:rules Loading rule 'no-lone-blocks' (remaining=220)
2020-09-01T19:40:18.379Z eslint:rules Loading rule 'no-lonely-if' (remaining=219)
2020-09-01T19:40:18.379Z eslint:rules Loading rule 'no-loss-of-precision' (remaining=218)
2020-09-01T19:40:18.379Z eslint:rules Loading rule 'no-loop-func' (remaining=217)
2020-09-01T19:40:18.380Z eslint:rules Loading rule 'no-misleading-character-class' (remaining=216)
2020-09-01T19:40:18.381Z eslint:rules Loading rule 'no-multi-assign' (remaining=215)
2020-09-01T19:40:18.381Z eslint:rules Loading rule 'no-new-func' (remaining=214)
2020-09-01T19:40:18.381Z eslint:rules Loading rule 'no-new-object' (remaining=213)
2020-09-01T19:40:18.382Z eslint:rules Loading rule 'no-new-symbol' (remaining=212)
2020-09-01T19:40:18.382Z eslint:rules Loading rule 'no-new-wrappers' (remaining=211)
2020-09-01T19:40:18.382Z eslint:rules Loading rule 'no-obj-calls' (remaining=210)
2020-09-01T19:40:18.383Z eslint:rules Loading rule 'no-octal' (remaining=209)
2020-09-01T19:40:18.383Z eslint:rules Loading rule 'no-octal-escape' (remaining=208)
2020-09-01T19:40:18.383Z eslint:rules Loading rule 'no-proto' (remaining=207)
2020-09-01T19:40:18.384Z eslint:rules Loading rule 'no-prototype-builtins' (remaining=206)
2020-09-01T19:40:18.384Z eslint:rules Loading rule 'no-redeclare' (remaining=205)
2020-09-01T19:40:18.385Z eslint:rules Loading rule 'no-regex-spaces' (remaining=204)
2020-09-01T19:40:18.385Z eslint:rules Loading rule 'no-restricted-globals' (remaining=203)
2020-09-01T19:40:18.387Z eslint:rules Loading rule 'no-return-assign' (remaining=202)
2020-09-01T19:40:18.387Z eslint:rules Loading rule 'no-return-await' (remaining=201)
2020-09-01T19:40:18.387Z eslint:rules Loading rule 'no-script-url' (remaining=200)
2020-09-01T19:40:18.388Z eslint:rules Loading rule 'no-self-assign' (remaining=199)
2020-09-01T19:40:18.389Z eslint:rules Loading rule 'no-self-compare' (remaining=198)
2020-09-01T19:40:18.389Z eslint:rules Loading rule 'no-sequences' (remaining=197)
2020-09-01T19:40:18.389Z eslint:rules Loading rule 'no-setter-return' (remaining=196)
2020-09-01T19:40:18.390Z eslint:rules Loading rule 'no-shadow' (remaining=195)
2020-09-01T19:40:18.391Z eslint:rules Loading rule 'no-shadow-restricted-names' (remaining=194)
2020-09-01T19:40:18.391Z eslint:rules Loading rule 'no-sparse-arrays' (remaining=193)
2020-09-01T19:40:18.391Z eslint:rules Loading rule 'no-this-before-super' (remaining=192)
2020-09-01T19:40:18.392Z eslint:rules Loading rule 'no-throw-literal' (remaining=191)
2020-09-01T19:40:18.392Z eslint:rules Loading rule 'no-undef' (remaining=190)
2020-09-01T19:40:18.393Z eslint:rules Loading rule 'no-unexpected-multiline' (remaining=189)
2020-09-01T19:40:18.393Z eslint:rules Loading rule 'no-unmodified-loop-condition' (remaining=188)
2020-09-01T19:40:18.394Z eslint:rules Loading rule 'no-unneeded-ternary' (remaining=187)
2020-09-01T19:40:18.395Z eslint:rules Loading rule 'no-unreachable' (remaining=186)
2020-09-01T19:40:18.395Z eslint:rules Loading rule 'no-unreachable-loop' (remaining=185)
2020-09-01T19:40:18.396Z eslint:rules Loading rule 'no-unsafe-finally' (remaining=184)
2020-09-01T19:40:18.396Z eslint:rules Loading rule 'no-unsafe-negation' (remaining=183)
2020-09-01T19:40:18.397Z eslint:rules Loading rule 'no-unused-expressions' (remaining=182)
2020-09-01T19:40:18.398Z eslint:rules Loading rule 'no-unused-labels' (remaining=181)
2020-09-01T19:40:18.398Z eslint:rules Loading rule 'no-unused-vars' (remaining=180)
2020-09-01T19:40:18.400Z eslint:rules Loading rule 'no-useless-backreference' (remaining=179)
2020-09-01T19:40:18.401Z eslint:rules Loading rule 'no-useless-call' (remaining=178)
2020-09-01T19:40:18.401Z eslint:rules Loading rule 'no-useless-catch' (remaining=177)
2020-09-01T19:40:18.402Z eslint:rules Loading rule 'no-useless-computed-key' (remaining=176)
2020-09-01T19:40:18.402Z eslint:rules Loading rule 'no-useless-concat' (remaining=175)
2020-09-01T19:40:18.403Z eslint:rules Loading rule 'no-useless-constructor' (remaining=174)
2020-09-01T19:40:18.403Z eslint:rules Loading rule 'no-useless-escape' (remaining=173)
2020-09-01T19:40:18.403Z eslint:rules Loading rule 'no-useless-rename' (remaining=172)
2020-09-01T19:40:18.405Z eslint:rules Loading rule 'no-useless-return' (remaining=171)
2020-09-01T19:40:18.405Z eslint:rules Loading rule 'no-var' (remaining=170)
2020-09-01T19:40:18.405Z eslint:rules Loading rule 'no-void' (remaining=169)
2020-09-01T19:40:18.406Z eslint:rules Loading rule 'no-warning-comments' (remaining=168)
2020-09-01T19:40:18.407Z eslint:rules Loading rule 'no-with' (remaining=167)
2020-09-01T19:40:18.407Z eslint:rules Loading rule 'object-shorthand' (remaining=166)
2020-09-01T19:40:18.409Z eslint:rules Loading rule 'one-var' (remaining=165)
2020-09-01T19:40:18.410Z eslint:rules Loading rule 'prefer-const' (remaining=164)
2020-09-01T19:40:18.411Z eslint:rules Loading rule 'prefer-destructuring' (remaining=163)
2020-09-01T19:40:18.413Z eslint:rules Loading rule 'prefer-exponentiation-operator' (remaining=162)
2020-09-01T19:40:18.414Z eslint:rules Loading rule 'prefer-numeric-literals' (remaining=161)
2020-09-01T19:40:18.414Z eslint:rules Loading rule 'prefer-object-spread' (remaining=160)
2020-09-01T19:40:18.414Z eslint:rules Loading rule 'prefer-promise-reject-errors' (remaining=159)
2020-09-01T19:40:18.415Z eslint:rules Loading rule 'prefer-regex-literals' (remaining=158)
2020-09-01T19:40:18.416Z eslint:rules Loading rule 'prefer-rest-params' (remaining=157)
2020-09-01T19:40:18.416Z eslint:rules Loading rule 'prefer-spread' (remaining=156)
2020-09-01T19:40:18.417Z eslint:rules Loading rule 'prefer-template' (remaining=155)
2020-09-01T19:40:18.417Z eslint:rules Loading rule 'radix' (remaining=154)
2020-09-01T19:40:18.418Z eslint:rules Loading rule 'require-unicode-regexp' (remaining=153)
2020-09-01T19:40:18.418Z eslint:rules Loading rule 'require-yield' (remaining=152)
2020-09-01T19:40:18.418Z eslint:rules Loading rule 'symbol-description' (remaining=151)
2020-09-01T19:40:18.419Z eslint:rules Loading rule 'use-isnan' (remaining=150)
2020-09-01T19:40:18.420Z eslint:rules Loading rule 'valid-typeof' (remaining=149)
2020-09-01T19:40:18.468Z eslint:rules Loading rule 'brace-style' (remaining=148)
2020-09-01T19:40:18.469Z eslint:rules Loading rule 'comma-spacing' (remaining=147)
2020-09-01T19:40:18.469Z eslint:rules Loading rule 'func-call-spacing' (remaining=146)
2020-09-01T19:40:18.469Z eslint:rules Loading rule 'indent' (remaining=145)
2020-09-01T19:40:18.469Z eslint:rules Loading rule 'init-declarations' (remaining=144)
2020-09-01T19:40:18.470Z eslint:rules Loading rule 'keyword-spacing' (remaining=143)
2020-09-01T19:40:18.470Z eslint:rules Loading rule 'no-extra-parens' (remaining=142)
2020-09-01T19:40:18.470Z eslint:rules Loading rule 'no-extra-semi' (remaining=141)
2020-09-01T19:40:18.471Z eslint:rules Loading rule 'no-magic-numbers' (remaining=140)
2020-09-01T19:40:18.472Z eslint:rules Loading rule 'no-use-before-define' (remaining=139)
2020-09-01T19:40:18.472Z eslint:rules Loading rule 'quotes' (remaining=138)
2020-09-01T19:40:18.472Z eslint:rules Loading rule 'semi' (remaining=137)
2020-09-01T19:40:18.472Z eslint:rules Loading rule 'space-before-function-paren' (remaining=136)
2020-09-01T19:40:18.478Z eslint:rules Loading rule 'require-await' (remaining=135)
2020-09-01T19:40:18.481Z eslint:cascading-config-array-factory Configuration was determined: ConfigArray(7) [ { type: 'config', name: 'DefaultIgnorePattern', filePath: '', criteria: null, env: undefined, globals: undefined, ignorePattern: IgnorePattern { patterns: [Array], basePath: '/home/gabriel/Repos/leapfrog', loose: false }, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { type: 'config', name: '.eslintrc.js » @gcangussu/eslint-config', filePath: '/home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { ecmaVersion: 2021, sourceType: 'module', project: [Array] }, plugins: { unicorn: [Object], import: [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'array-callback-return': [Array], 'block-scoped-var': 'error', 'class-methods-use-this': 'error', 'consistent-return': 'warn', 'constructor-super': 'error', 'default-case': 'warn', 'default-param-last': 'error', eqeqeq: [Array], 'for-direction': 'error', 'func-names': [Array], 'getter-return': 'error', 'grouped-accessor-pairs': 'error', 'guard-for-in': 'error', 'new-cap': [Array], 'new-parens': 'error', 'no-alert': 'warn', 'no-array-constructor': 'error', 'no-async-promise-executor': 'error', 'no-buffer-constructor': 'error', 'no-caller': 'error', 'no-case-declarations': 'error', 'no-class-assign': 'error', 'no-compare-neg-zero': 'error', 'no-cond-assign': 'error', 'no-console': 'warn', 'no-const-assign': 'error', 'no-constant-condition': [Array], 'no-constructor-return': 'error', 'no-control-regex': 'error', 'no-debugger': 'error', 'no-delete-var': 'error', 'no-dupe-args': 'error', 'no-dupe-class-members': 'error', 'no-dupe-else-if': 'error', 'no-dupe-keys': 'error', 'no-duplicate-case': 'error', 'no-duplicate-imports': 'error', 'no-else-return': 'error', 'no-empty': 'error', 'no-empty-character-class': 'error', 'no-empty-function': 'error', 'no-empty-pattern': 'error', 'no-eval': 'error', 'no-ex-assign': 'error', 'no-extend-native': 'error', 'no-extra-bind': 'error', 'no-extra-boolean-cast': 'error', 'no-extra-label': 'error', 'no-fallthrough': 'error', 'no-floating-decimal': 'error', 'no-func-assign': 'error', 'no-global-assign': 'error', 'no-implicit-coercion': 'error', 'no-implied-eval': 'error', 'no-import-assign': 'error', 'no-inner-declarations': [Array], 'no-invalid-regexp': 'error', 'no-irregular-whitespace': 'error', 'no-iterator': 'error', 'no-label-var': 'error', 'no-labels': 'error', 'no-lone-blocks': 'error', 'no-lonely-if': 'error', 'no-loss-of-precision': 'error', 'no-loop-func': 'error', 'no-misleading-character-class': 'error', 'no-multi-assign': 'error', 'no-new-func': 'error', 'no-new-object': 'error', 'no-new-symbol': 'error', 'no-new-wrappers': 'error', 'no-obj-calls': 'error', 'no-octal': 'error', 'no-octal-escape': 'error', 'no-proto': 'error', 'no-prototype-builtins': 'error', 'no-redeclare': 'error', 'no-regex-spaces': 'error', 'no-restricted-globals': [Array], 'no-return-assign': 'error', 'no-return-await': 'error', 'no-script-url': 'error', 'no-self-assign': 'error', 'no-self-compare': 'error', 'no-sequences': 'error', 'no-setter-return': 'error', 'no-shadow': 'warn', 'no-shadow-restricted-names': 'error', 'no-sparse-arrays': 'error', 'no-this-before-super': 'error', 'no-throw-literal': 'error', 'no-undef': 'error', 'no-unexpected-multiline': 'error', 'no-unmodified-loop-condition': 'error', 'no-unneeded-ternary': 'error', 'no-unreachable': 'error', 'no-unreachable-loop': 'error', 'no-unsafe-finally': 'error', 'no-unsafe-negation': 'error', 'no-unused-expressions': [Array], 'no-unused-labels': 'error', 'no-unused-vars': 'error', 'no-useless-backreference': 'error', 'no-useless-call': 'error', 'no-useless-catch': 'error', 'no-useless-computed-key': 'error', 'no-useless-concat': 'error', 'no-useless-constructor': 'error', 'no-useless-escape': 'error', 'no-useless-rename': 'error', 'no-useless-return': 'error', 'no-var': 'error', 'no-void': 'error', 'no-warning-comments': 'warn', 'no-with': 'error', 'object-shorthand': [Array], 'one-var': [Array], 'prefer-const': 'error', 'prefer-destructuring': [Array], 'prefer-exponentiation-operator': 'error', 'prefer-numeric-literals': 'error', 'prefer-object-spread': 'error', 'prefer-promise-reject-errors': 'error', 'prefer-regex-literals': 'error', 'prefer-rest-params': 'error', 'prefer-spread': 'error', 'prefer-template': 'warn', radix: 'error', 'require-unicode-regexp': 'error', 'require-yield': 'error', 'symbol-description': 'error', 'use-isnan': [Array], 'valid-typeof': [Array], 'unicorn/custom-error-definition': 'error', 'unicorn/error-message': 'error', 'unicorn/explicit-length-check': 'warn', 'unicorn/no-abusive-eslint-disable': 'error', 'unicorn/no-array-instanceof': 'error', 'unicorn/no-console-spaces': 'warn', 'unicorn/prefer-event-key': 'error', 'unicorn/prefer-flat-map': 'warn', 'unicorn/prefer-includes': 'warn', 'unicorn/prefer-modern-dom-apis': 'warn', 'unicorn/prefer-negative-index': 'warn', 'unicorn/prefer-node-append': 'warn', 'unicorn/prefer-node-remove': 'warn', 'unicorn/prefer-reflect-apply': 'warn', 'unicorn/prefer-set-has': 'warn', 'unicorn/prefer-starts-ends-with': 'warn', 'unicorn/prefer-string-slice': 'warn', 'unicorn/prefer-text-content': 'warn', 'unicorn/prefer-trim-start-end': 'error', 'unicorn/throw-new-error': 'error', 'import/no-unresolved': [Array], 'import/named': 'error', 'import/default': 'error', 'import/namespace': [Array], 'import/no-dynamic-require': 'error', 'import/no-webpack-loader-syntax': 'error', 'import/no-self-import': 'error', 'import/no-cycle': [Array], 'import/no-useless-path-segments': [Array], 'import/export': 'error', 'import/no-extraneous-dependencies': [Array], 'import/newline-after-import': 'error', 'import/no-unassigned-import': [Array], 'import/no-anonymous-default-export': 'error' }, settings: { 'import/extensions': [Array], 'import/parsers': [Object], 'import/resolver': [Object] } }, { type: 'config', name: '.eslintrc.js » @gcangussu/eslint-config#overrides[0]', filePath: '/home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/index.js', criteria: { includes: [Array], excludes: null, basePath: '/home/gabriel/Repos/leapfrog' }, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '/home/gabriel/Repos/leapfrog/node_modules/@typescript-eslint/parser/dist/index.js', id: '@typescript-eslint/parser', importerName: '.eslintrc.js » @gcangussu/eslint-config#overrides[0]', importerPath: '/home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/index.js' }, parserOptions: undefined, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/adjacent-overload-signatures': 'error', '@typescript-eslint/ban-types': 'error', '@typescript-eslint/naming-convention': [Array], '@typescript-eslint/no-empty-interface': [Array], '@typescript-eslint/no-explicit-any': [Array], '@typescript-eslint/no-extra-non-null-assertion': 'error', '@typescript-eslint/no-extraneous-class': 'error', '@typescript-eslint/no-inferrable-types': 'warn', '@typescript-eslint/no-invalid-void-type': 'error', '@typescript-eslint/no-misused-new': 'error', '@typescript-eslint/no-namespace': 'error', '@typescript-eslint/no-non-null-asserted-optional-chain': 'error', '@typescript-eslint/prefer-as-const': 'warn', '@typescript-eslint/triple-slash-reference': 'error', 'brace-style': 'off', '@typescript-eslint/brace-style': 'off', 'comma-spacing': 'off', '@typescript-eslint/comma-spacing': 'off', 'default-param-last': 'off', '@typescript-eslint/default-param-last': 'error', 'func-call-spacing': 'off', '@typescript-eslint/func-call-spacing': 'off', indent: 'off', '@typescript-eslint/indent': 'off', 'init-declarations': 'off', '@typescript-eslint/init-declarations': 'off', 'keyword-spacing': 'off', '@typescript-eslint/keyword-spacing': 'off', 'no-array-constructor': 'off', '@typescript-eslint/no-array-constructor': 'error', 'no-dupe-class-members': 'off', '@typescript-eslint/no-dupe-class-members': 'error', 'no-empty-function': 'off', '@typescript-eslint/no-empty-function': 'error', 'no-extra-parens': 'off', '@typescript-eslint/no-extra-parens': 'off', 'no-extra-semi': 'off', '@typescript-eslint/no-extra-semi': 'off', 'no-loss-of-precision': 'off', '@typescript-eslint/no-loss-of-precision': 'error', 'no-magic-numbers': 'off', '@typescript-eslint/no-magic-numbers': 'off', 'no-shadow': 'off', '@typescript-eslint/no-shadow': 'warn', 'no-unused-expressions': 'off', '@typescript-eslint/no-unused-expressions': [Array], 'no-unused-vars': 'off', '@typescript-eslint/no-unused-vars': 'off', 'no-use-before-define': 'off', '@typescript-eslint/no-use-before-define': 'off', 'no-useless-constructor': 'off', '@typescript-eslint/no-useless-constructor': 'error', quotes: 'off', '@typescript-eslint/quotes': 'off', semi: 'off', '@typescript-eslint/semi': 'off', 'space-before-function-paren': 'off', '@typescript-eslint/space-before-function-paren': 'off', 'getter-return': 'off', 'no-dupe-args': 'off', 'no-dupe-keys': 'off', 'no-unreachable': 'off', 'valid-typeof': 'off', 'no-const-assign': 'off', 'no-new-symbol': 'off', 'no-this-before-super': 'off', 'no-undef': 'off', 'no-redeclare': 'off', 'consistent-return': 'off', 'import/named': 'off', 'import/default': 'off', 'import/namespace': 'off', 'import/no-named-as-default-member': 'off', '@typescript-eslint/await-thenable': 'error', '@typescript-eslint/no-base-to-string': 'error', '@typescript-eslint/no-floating-promises': 'error', '@typescript-eslint/no-for-in-array': 'error', '@typescript-eslint/no-implied-eval': 'error', '@typescript-eslint/no-misused-promises': 'error', '@typescript-eslint/no-throw-literal': 'error', '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error', '@typescript-eslint/no-unnecessary-condition': 'error', '@typescript-eslint/no-unnecessary-qualifier': 'warn', '@typescript-eslint/no-unnecessary-type-assertion': 'warn', '@typescript-eslint/no-unsafe-assignment': 'error', '@typescript-eslint/no-unsafe-call': 'error', '@typescript-eslint/no-unsafe-member-access': 'error', '@typescript-eslint/no-unsafe-return': 'error', '@typescript-eslint/prefer-includes': 'warn', '@typescript-eslint/prefer-reduce-type-parameter': 'warn', '@typescript-eslint/prefer-regexp-exec': 'error', '@typescript-eslint/prefer-string-starts-ends-with': 'error', '@typescript-eslint/promise-function-async': 'error', '@typescript-eslint/require-array-sort-compare': 'error', '@typescript-eslint/restrict-plus-operands': [Array], '@typescript-eslint/restrict-template-expressions': [Array], '@typescript-eslint/strict-boolean-expressions': 'error', '@typescript-eslint/switch-exhaustiveness-check': 'error', 'require-await': 'off', '@typescript-eslint/require-await': 'off', 'no-return-await': 'off', '@typescript-eslint/return-await': 'error' }, settings: undefined }, { type: 'config', name: '.eslintrc.js » @gcangussu/eslint-config/quick', filePath: '/home/gabriel/Repos/leapfrog/node_modules/@gcangussu/eslint-config/quick.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { project: null }, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'import/no-cycle': 'off', 'import/no-deprecated': 'off', 'import/no-named-as-default': 'off', 'import/no-named-as-default-member': 'off', 'import/no-unused-modules': 'off', '@typescript-eslint/await-thenable': 'off', '@typescript-eslint/no-base-to-string': 'off', '@typescript-eslint/no-floating-promises': 'off', '@typescript-eslint/no-for-in-array': 'off', '@typescript-eslint/no-implied-eval': 'off', '@typescript-eslint/no-misused-promises': 'off', '@typescript-eslint/no-throw-literal': 'off', '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'off', '@typescript-eslint/no-unnecessary-condition': 'off', '@typescript-eslint/no-unnecessary-qualifier': 'off', '@typescript-eslint/no-unnecessary-type-assertion': 'off', '@typescript-eslint/no-unsafe-assignment': 'off', '@typescript-eslint/no-unsafe-call': 'off', '@typescript-eslint/no-unsafe-member-access': 'off', '@typescript-eslint/no-unsafe-return': 'off', '@typescript-eslint/prefer-includes': 'off', '@typescript-eslint/prefer-reduce-type-parameter': 'off', '@typescript-eslint/prefer-regexp-exec': 'off', '@typescript-eslint/prefer-string-starts-ends-with': 'off', '@typescript-eslint/promise-function-async': 'off', '@typescript-eslint/require-array-sort-compare': 'off', '@typescript-eslint/restrict-plus-operands': 'off', '@typescript-eslint/restrict-template-expressions': 'off', '@typescript-eslint/strict-boolean-expressions': 'off', '@typescript-eslint/switch-exhaustiveness-check': 'off', '@typescript-eslint/require-await': 'off', '@typescript-eslint/return-await': 'off' }, settings: undefined }, { type: 'config', name: '.eslintrc.js', filePath: '/home/gabriel/Repos/leapfrog/.eslintrc.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { ecmaVersion: 2021, project: null }, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: true, rules: { 'import/no-unresolved': [Array] }, settings: undefined }, { type: 'config', name: '--config', filePath: '/home/gabriel/Repos/leapfrog/.eslintrc.test.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { ecmaVersion: 2021, project: null }, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: true, rules: { '@typescript-eslint/ban-types': 'error' }, settings: undefined }, { type: 'ignore', name: '.eslintignore', filePath: '/home/gabriel/Repos/leapfrog/.eslintignore', criteria: null, env: undefined, globals: undefined, ignorePattern: IgnorePattern { patterns: [Array], basePath: '/home/gabriel/Repos/leapfrog', loose: true }, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined } ] on /home/gabriel/Repos/leapfrog
2020-09-01T19:40:18.482Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '/home/gabriel/Repos/leapfrog', loose: false }, IgnorePattern { patterns: [ 'node_modules/', 'dist/', 'build/', '/backend' ], basePath: '/home/gabriel/Repos/leapfrog', loose: true } ]
2020-09-01T19:40:18.483Z eslint:ignore-pattern   processed: { basePath: '/home/gabriel/Repos/leapfrog', patterns: [ '/**/node_modules/*', 'node_modules/', 'dist/', 'build/', '/backend' ] }
2020-09-01T19:40:18.483Z eslint:ignore-pattern Check {
  filePath: '/home/gabriel/Repos/leapfrog/test.ts',
  dot: false,
  relativePath: 'test.ts',
  result: false
}
2020-09-01T19:40:18.483Z eslint:cli-engine Lint /home/gabriel/Repos/leapfrog/test.ts
2020-09-01T19:40:18.484Z eslint:linter Linting code for /home/gabriel/Repos/leapfrog/test.ts (pass 1)
2020-09-01T19:40:18.484Z eslint:linter Verify
2020-09-01T19:40:18.484Z eslint:linter With ConfigArray: /home/gabriel/Repos/leapfrog/test.ts
2020-09-01T19:40:18.500Z eslint:linter Parsing error: Invalid value for lib provided: es2021
Error: Invalid value for lib provided: es2021
    at Referencer.populateGlobalsFromLib (/home/gabriel/Repos/leapfrog/node_modules/@typescript-eslint/scope-manager/dist/referencer/Referencer.js:66:23)
    at Referencer.Program (/home/gabriel/Repos/leapfrog/node_modules/@typescript-eslint/scope-manager/dist/referencer/Referencer.js:350:14)
    at Referencer.visit (/home/gabriel/Repos/leapfrog/node_modules/@typescript-eslint/scope-manager/dist/referencer/VisitorBase.js:72:28)
    at Object.analyze (/home/gabriel/Repos/leapfrog/node_modules/@typescript-eslint/scope-manager/dist/analyze.js:43:16)
    at Object.parseForESLint (/home/gabriel/Repos/leapfrog/node_modules/@typescript-eslint/parser/dist/parser.js:105:42)
    at parse (/home/gabriel/Repos/leapfrog/node_modules/eslint/lib/linter/linter.js:655:22)
    at Linter._verifyWithoutProcessors (/home/gabriel/Repos/leapfrog/node_modules/eslint/lib/linter/linter.js:1127:33)
    at Linter._verifyWithConfigArray (/home/gabriel/Repos/leapfrog/node_modules/eslint/lib/linter/linter.js:1271:21)
    at Linter.verify (/home/gabriel/Repos/leapfrog/node_modules/eslint/lib/linter/linter.js:1226:25)
    at Linter.verifyAndFix (/home/gabriel/Repos/leapfrog/node_modules/eslint/lib/linter/linter.js:1416:29)
2020-09-01T19:40:18.500Z eslint:linter Generating fixed text for /home/gabriel/Repos/leapfrog/test.ts (pass 1)
2020-09-01T19:40:18.500Z eslint:source-code-fixer Applying fixes
2020-09-01T19:40:18.501Z eslint:source-code-fixer shouldFix parameter was false, not attempting fixes
2020-09-01T19:40:18.501Z eslint:file-enumerator Complete iterating files: ["./test.ts"]
2020-09-01T19:40:18.501Z eslint:cli-engine Linting complete in: 675ms

/home/gabriel/Repos/leapfrog/test.ts
  0:0  error  Parsing error: Invalid value for lib provided: es2021

✖ 1 problem (1 error, 0 warnings)

```

</p>
</details>

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.0.1` |
| `TypeScript`                | `4.0.2` |
| `ESLint`                    | `7.8.0` |
| `node`                      | `14.8.0` |

--- a/packages/scope-manager/src/analyze.ts
+++ b/packages/scope-manager/src/analyze.ts
@@ -2,6 +2,7 @@ import { TSESTree, EcmaVersion, Lib } from '@typescript-eslint/types';
 import { visitorKeys } from '@typescript-eslint/visitor-keys';
 import { Referencer, ReferencerOptions } from './referencer';
 import { ScopeManager } from './ScopeManager';
+import { lib as TSLibraries } from './lib';
 
 ////////////////////////////////////////////////////
 // MAKE SURE THIS IS KEPT IN SYNC WITH THE README //
@@ -61,12 +62,10 @@ function mapEcmaVersion(version: EcmaVersion | undefined): Lib {
     return 'es5';
   }
 
-  if (version > 2000) {
-    return `es${version}` as Lib;
-  }
+  const year = version > 2000 ? version : 2015 + (version - 6);
+  const lib = `es${year}`;
 
-  const year = 2015 + (version - 6);
-  return `es${year}` as Lib;
+  return lib in TSLibraries ? (lib as Lib) : year > 2020 ? 'esnext' : 'es5';
 }
 
 /**

--- a/packages/scope-manager/tests/eslint-scope/map-ecma-version.test.ts
+++ b/packages/scope-manager/tests/eslint-scope/map-ecma-version.test.ts
@@ -16,17 +16,26 @@ describe('ecma version mapping', () => {
     expectMapping(2, 'es5');
   });
 
-  it("should map to 'es{year}' when supported", () => {
+  it("should map to 'es{year}' when supported and >= 6", () => {
     expectMapping(2015, 'es2015');
     expectMapping(6, 'es2015');
     expectMapping(2020, 'es2020');
     expectMapping(11, 'es2020');
   });
+
+  it("should map to 'es5' when 5 or 3", () => {
+    expectMapping(5, 'es5');
+    expectMapping(3, 'es5');
+  });
+
+  it("should map to 'es2018' when undefined", () => {
+    expectMapping(undefined, 'es2018');
+  });
 });
 
 const fakeNode = ({} as unknown) as TSESTree.Node;
 
-function expectMapping(ecmaVersion: number, lib: Lib): void {
+function expectMapping(ecmaVersion: number | undefined, lib: Lib): void {
   (Referencer as jest.Mock).mockClear();
   analyze(fakeNode, { ecmaVersion: ecmaVersion as EcmaVersion });
   expect(Referencer).toHaveBeenCalledWith(

--- a/packages/scope-manager/tests/eslint-scope/map-ecma-version.test.ts
+++ b/packages/scope-manager/tests/eslint-scope/map-ecma-version.test.ts
@@ -1,0 +1,36 @@
+import { analyze } from '../../src/analyze';
+import { Referencer } from '../../src/referencer';
+import { TSESTree, EcmaVersion, Lib } from '@typescript-eslint/types';
+
+jest.mock('../../src/referencer');
+jest.mock('../../src/ScopeManager');
+
+describe('ecma version mapping', () => {
+  it("should map to 'esnext' when unsuported and new", () => {
+    expectMapping(2042, 'esnext');
+    expectMapping(42, 'esnext');
+  });
+
+  it("should map to 'es5' when unsuported and old", () => {
+    expectMapping(2002, 'es5');
+    expectMapping(2, 'es5');
+  });
+
+  it("should map to 'es{year}' when supported", () => {
+    expectMapping(2015, 'es2015');
+    expectMapping(6, 'es2015');
+    expectMapping(2020, 'es2020');
+    expectMapping(11, 'es2020');
+  });
+});
+
+const fakeNode = ({} as unknown) as TSESTree.Node;
+
+function expectMapping(ecmaVersion: number, lib: Lib): void {
+  (Referencer as jest.Mock).mockClear();
+  analyze(fakeNode, { ecmaVersion: ecmaVersion as EcmaVersion });
+  expect(Referencer).toHaveBeenCalledWith(
+    expect.objectContaining({ lib: [lib] }),
+    expect.any(Object),
+  );
+}

